_posts/2023-02-01-nathanlewis2001-limited-direct0execution-meme.md

```diff
@@ -0,0 +1,7 @@
+---
+title:  Power (Kernel) Mode
+author: nate
+image: OSCh6Meme5.jpeg
+chapter: 6
+---
+When user mode calls the trap, that promotes it into kernel mode where it can perform all operations.
```

Comment:
I really want to like this one, but it is incorrect. User/kernel modes are states of the processor that change when a trap or return-from-trap (CPU instruction) is executed. A kernel process is simultaneously switched onto the CPU. That trap is part of a system call that a process running in user mode calls. Nothing gets promoted here. If the user process running the system call is promoted, that would be a "Very Bad Idea" for security.